addon/components/polaris-text-field.js

```diff
@@ -0,0 +1,520 @@
+
+    return `Polaris-TextField ${ valueClass } ${ disabledClass } ${ readOnlyClass } ${ errorClass } ${ multilineClass } ${ focusClass }`;
+  }),
+
+  ariaDescribedBy: computed('error', 'helpText', 'id', function() {
+    let { error, helpText, id } = this.getProperties('error', 'helpText', 'id');
+    let describedBy = [];
+
+    if (error) {
+      describedBy.push(`${ id }Error`);
+    }
+
+    if (helpText) {
+      describedBy.push(`${ id }HelpText`);
+    }
+
+    return describedBy.join(' ');
+  }),
+
+  ariaLabelledBy: computed('id', function() {
+    return `${ this.get('id') }Label`;
+  }),
+
+  ariaInvalid: computed('error', function() {
+    return !!this.get('error');
+  }),
+
+  inputType: computed('type', function() {
+    let type = this.get('type');
+
+    return type === 'currency' ? 'text' : type;
+  }),
+
+  input: computed('multiline', function() {
+    let multiline = this.get('multiline');
+
+    return multiline ? 'textarea' : 'input';
+  }),
+
+  minimumLines: computed('multiline', function() {
+    let multiline = this.get('multiline');
+
+    return (typeof multiline === 'number') ? multiline : 1;
+  }),
+
+  heightStyle: computed('height', 'multiline', function() {
+    let { height, multiline } = this.getProperties('height', 'multiline');
+
+    return (height && multiline) ? htmlSafe(`height: ${ height }px`) : '';
+  }),
+
+  init() {
+    this._super(...arguments);
+
+    let { id, type } = this.getProperties('id', 'type');
+    let getUniqueID = createUniqueIDFactory('TextField');
+
+    assert(`ember-polaris::polaris-text-field - ${ type } is not a valid type.`, allowedTypes.indexOf(type) > -1);
+
+    id = id || getUniqueID();
+
+    this.setProperties({
+      height: null,
+      focus: false,
+      id
+    });
+
+    next(this, ()=> {
+      this.addValueListener();
+    });
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    let { input, focused } = this.getProperties('input', 'focused');
+
+    if (input && focused) {
+      input.focus();
+    }
+  },
+
+  willDestroyElement() {
+    this.removeValueListener();
+  },
+
+  destroy() {
+    runDisposables(this);
+
+    this._super(...arguments);
+  },
+
+  actions: {
+    handleNumberChange(steps) {
+      let {
+        id,
+        onChange,
+        value,
+        step,
+        min,
+        max,
+      } = this.getProperties('id', 'onChange', 'value', 'step', 'min', 'max');
+
+      step = step || 1;
+      min = min || -Infinity;
+      max = max || Infinity;
+
+      let numericValue = value ? parseFloat(value) : 0;
+
+      if (isNaN(numericValue)) {
+        return;
+      }
+
+      // Making sure the new value has the same length of decimal places as the
+      // step / value has.
+      let decimalPlaces = Math.max(dpl(numericValue), dpl(step));
+      let newValue = Math.min(max, Math.max(numericValue + steps * step, min));
+
+      onChange(String(newValue.toFixed(decimalPlaces)), id);
```

Comment:
I know this is duplicated from the React code, but doesn't `toFixed` return a string anyway? 😕

Reply:
Hm yeah it would. Weird, I'll get rid of the `String` call.

Reply:
I'm good with it staying there since it's come from the React code, up to you 🤷‍♂️